applications/CompressiblePotentialFlowApplication/custom_response_functions/adjoint_potential_response_function.cpp

```diff
@@ -40,9 +40,15 @@ namespace Kratos
             double delta = ResponseSettings["step_size"].GetDouble();
             mDelta = delta;
         }
+        else if (gradient_mode == "analytic")
```

Comment:
minor comment:
It is a bit weird that we define how we derive the elements in the "response_function_settings", but I understand that some refactoring of the adjoint_solver would be needed otherwise.

Reply:
Hi there!

I am definitely not an expert in adjoints and I most probably I made a mistake here, but I do not understand why you find it weird that we define the gradient mode here. What other alternative are you thinking of? Where would you define it?

As far as I remember when I was implementing this, I understood that the variables gradient_mode and mGradientMode are precisely there to indicate the way the sensitivities are being computed: finite differences, semi analytic or analytic. This also complies to how the element is selected according to the response_function_settings in the potential_flow_adjoint_solver.py. Am I missing something here?

Reply:
Sorry for not answering earlier! 

What I meant is that what is changing from analytical/semi-analytical is the element not the respone function (which is analytical right now). 

Actually i think the fact that there's a "semi_analytic" option right now is a mistake from my first PR that went through (since we are using the analytical approach here with the potentail jump). If you can I would remove the semi_analytic option and just leave the analytical one.

Reply:
Ah ok! Now I understand what you meant. My confusion was coming because there is only one setting deciding on the gradient mode. As far as I understood now you are suggesting that we define two different settings, one that defines the gradient mode of the response function and another one that defines the gradient mode of the element, is that right?

I think in practice people just look at the sensitivities as a whole and as soon as one term is computed using finite-differences the whole sensitivities are said to be computed semi-analytically. However what you propose is a good idea.

In that case, shall we pass the step size directly to the element via rCurrentProcessInfo? Right now the step size is set in the response function, even though as you say the response function is computed analytically. However passing the step size directly to the element can lead to another confusion in the future when the response function is also computed using finite-differences... what do you think is the best approach?